File: app/src/main/java/joebruckner/lastpick/data/MovieViewHolder.java
package joebruckner.lastpick.data;

import android.support.annotation.NonNull;
import android.view.View;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import butterknife.Bind;
import butterknife.ButterKnife;
import joebruckner.lastpick.R;

public class MovieViewHolder {
    @Bind(R.id.title)
    TextView title;
    @Bind(R.id.summary)
    TextView summary;
    @Bind(R.id.cast)
    TextView cast;
    @Bind(R.id.year)
    TextView year;
    @Bind(R.id.rating)
    TextView rating;
    @Bind(R.id.length)
    TextView length;

    public MovieViewHolder(@NonNull View view) {
        ButterKnife.bind(this, view);
    }

    public void setData(Movie movie) {
        setTitle(movie.getTitle());
        setSummary(movie.getOverview());
        setCast(new ArrayList<>());
        setYear(movie.getReleaseDate());
        setRating(movie.getPopularity());
        setLength(movie.getRuntime());
    }

    public void setTitle(@NonNull String text) {
        title.setText(text);
    }

    public void setSummary(@NonNull String text) {
        summary.setText(text);
    }

    public void setCast(@NonNull List<Object> castList) {
        cast.setText("Cast: No acknowledgements found");
    }

    public void setYear(@NonNull String text) {
        year.setText(text.substring(0, 4));
    }

    public void setRating(double userRating) {
        rating.setText(Double.toString(userRating));
    }

    public void setLength(int runtime) {
        int hours = runtime / 60;
        int minutes = runtime % 60;
        StringBuilder text = new StringBuilder();
        text.append(hours);
        text.append(hours < 2 ? " hour " : " hours ");
        text.append(minutes);
        text.append(minutes < 2 ? " minute" : " minutes");
        length.setText(text);
    }
}

File: app/src/main/java/joebruckner/lastpick/UI/MovieViewHolder.kt
package joebruckner.lastpick.ui

import android.view.View
import android.widget.TextView
import joebruckner.lastpick.R
import joebruckner.lastpick.data.Credits
import joebruckner.lastpick.data.Movie

class MovieViewHolder(view: View) {
    //val title: TextView
    val summary: TextView
    val cast: TextView
    val year: TextView
    val mpaa: TextView
    val vote: TextView
    val length: TextView
    var movie: Movie? = null
        set(m: Movie?) {
           field = m
           summary.text    = m?.overview
           cast.text       = formatCredits(m?.credits)
           year.text       = m?.releaseDate?.substring(0,4)
           mpaa.text       = m?.getSimpleMpaa()
           vote.text       = formatVote(m?.voteAverage ?: 0.0)
           length.text     = formatLength(m?.runtime ?: 0)
        }

    init {
        //title   = view.findViewById(R.id.title)     as TextView
        summary = view.findViewById(R.id.overview)  as TextView
        cast    = view.findViewById(R.id.cast)      as TextView
        year    = view.findViewById(R.id.year)      as TextView
        mpaa  = view.findViewById(R.id.mpaa)        as TextView
        vote = view.findViewById(R.id.popularity)   as TextView
        length  = view.findViewById(R.id.runtime)   as TextView
    }

    fun formatCredits(credits: Credits?): String {
        if (credits == null) return ""

        val text = StringBuilder()
        val director = credits.getDirector()
        if (director != null)
            text.appendln("Directed by ${director.name}")
        for (i in 0..4) {
            if (credits.cast.size > i)
                text.append(
                        "\n${credits.cast[i].name}" +
                        " as " +
                        "${credits.cast[i].firstCharacter()}"
                )
        }
        return text.toString()
    }

    fun formatLength(runtime: Int): String {
        val hours = runtime / 60
        val minutes = runtime % 60
        val text = StringBuilder()
        text.append(hours)
        text.append(":")
        if (minutes < 10) text.append("0")
        text.append(minutes)
        return text.toString()
    }

    fun formatVote(vote: Double): String {
        return vote.toString()
    }
}